openfl/component/aggregator/aggregator.py

```diff
@@ -365,6 +366,14 @@ def get_aggregated_tensor(self, collaborator_name, tensor_name,
 
         nparray = self.tensor_db.get_tensor_from_cache(agg_tensor_key)
 
+        start_retrieving_time = time.time()
+        while(nparray is None):
+            self.logger.debug(f'Waiting for tensor_key {agg_tensor_key}')
+            time.sleep(5)
+            nparray = self.tensor_db.get_tensor_from_cache(agg_tensor_key)
+            if (time.time() - start_retrieving_time) > 60:
```

Comment:
I like this approach, but would suggest making the stop criteria (60 seconds) user configurable in the plan under aggregator settings. This will be useful for debugging as well as future use cases